src/Grpc.AspNetCore.Server/Internal/GrpcEventSource.cs

```diff
@@ -0,0 +1,159 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Grpc.Core;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class GrpcEventSource : EventSource
+    {
+        public static readonly GrpcEventSource Log = new GrpcEventSource();
```

Comment:
I'm not a fan of the static (even though it's the event source pattern), hosting adds the event source to the container.

Reply:
https://github.com/aspnet/AspNetCore/blob/3ce6c1dc5b022116ec9b13f9158e092323ac1158/src/Hosting/Hosting/src/Internal/HostingEventSource.cs#L13

No? Are you thinking of something else.

Reply:
https://github.com/aspnet/AspNetCore/blob/3ce6c1dc5b022116ec9b13f9158e092323ac1158/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L80-L84

There is a DiagnosticsSource injected via DI. gRPC isn't creating one of those on the server.

Reply:
Oh wait, that was another branch. Ignore me... 😄